src/lib.rs

```diff
@@ -54,6 +59,8 @@ use table::BucketState::{
     Full,
 };
 
+pub use adaptive_map::HashMapInterface;
```

Comment:
It seems like the inherent methods on `HashMap` could defer to `HashMapInterface`, no?